app/src/main/java/helper/EndlessRecyleOnScrollListenerStaggered.java

```diff
@@ -0,0 +1,56 @@
+package helper;
+
+import android.support.v7.widget.RecyclerView;
+import android.support.v7.widget.StaggeredGridLayoutManager;
+
+/**
+ * Created by Sukriti on 7/25/16.
+ */
+
+/**
+from a stackOverflow response - code not written by me
+ */
+
+ public abstract class EndlessRecyleOnScrollListenerStaggered extends RecyclerView.OnScrollListener {
+    public static String TAG = EndlessRecyleOnScrollListenerStaggered.class
+            .getSimpleName();
+
+    private boolean loading = true;
+    int pastVisibleItems, visibleItemCount, totalItemCount;
+
+    private int current_page = 1;
+
+    private StaggeredGridLayoutManager mStaggeredGridLayoutManager;
+
+    public EndlessRecyleOnScrollListenerStaggered(
+            StaggeredGridLayoutManager staggeredGridLayoutManager) {
+        this.mStaggeredGridLayoutManager = staggeredGridLayoutManager;
+    }
+
+    @Override
+    public void onScrolled(RecyclerView recyclerView, int dx, int dy) {
+        super.onScrolled(recyclerView, dx, dy);
+        visibleItemCount = mStaggeredGridLayoutManager.getChildCount();
+        totalItemCount = mStaggeredGridLayoutManager.getItemCount();
+        int[] firstVisibleItems = null;
+        firstVisibleItems = mStaggeredGridLayoutManager.findFirstVisibleItemPositions(firstVisibleItems);
+        if (firstVisibleItems != null && firstVisibleItems.length > 0) {
+            pastVisibleItems = firstVisibleItems[0];
+        }
+        if (loading) {
+            if ((visibleItemCount + pastVisibleItems) >= (totalItemCount)) {
```

Comment:
Good idea to use a guard to prevent loading until more items are loaded. Unfortunately, the code above will not always work because:
1. loading is set to true immediately after onLoadMore is called, hence is a little redundant since we can't guarantee that the item counts are updated immediately after the call to onLoadMore.
2. The real guard is whether visibleItem + pastVisibleItems >= totalItemCount, but as mentioned above, we can't guarantee when the layout manager updates its values.